syncer/syncer.go

```diff
@@ -218,17 +232,25 @@ func (s *Syncer) ban(p *Peer, err error) {
 			s.strikes[subnet]++
 		}
 		s.mu.Unlock()
-		if ban {
-			s.pm.Ban(subnet, 24*time.Hour, "too many strikes")
+		if !ban {
+			continue
+		} else if err := s.pm.Ban(subnet, 24*time.Hour, "too many strikes"); err != nil {
```

Comment:
can reuse the constant here already, maybe reuse `ErrPeerBanned` here and update the error message, I feel this error should be exported